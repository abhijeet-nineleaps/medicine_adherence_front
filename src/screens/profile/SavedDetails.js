import React from 'react';
import {Text, View} from 'react-native';
import styles from './profileStyles/ProfileStyles';
import {useFocusEffect} from '@react-navigation/native';
import AsyncStorage from '@react-native-async-storage/async-storage';

import Icon from 'react-native-vector-icons/FontAwesome5';
import OctIcon from 'react-native-vector-icons/Octicons';
import IonIcon from 'react-native-vector-icons/Ionicons';
import EntIcon from 'react-native-vector-icons/Entypo';

const SavedDetails = () => {
  const [bio, biostate] = React.useState('');
  const [contact, contactstate] = React.useState('');
  const [age, agestate] = React.useState('');
  const [weight, weightstate] = React.useState('');
  const [gender, genderstate] = React.useState('');
  const [ms, msstate] = React.useState('');
  const [blood, bloodstate] = React.useState('');

  useFocusEffect(() => {
    async function getuserdetail() {
      let sbio = await AsyncStorage.getItem('bio');
      let scontact = await AsyncStorage.getItem('contact');
      let sage = await AsyncStorage.getItem('age');
      let sweight = await AsyncStorage.getItem('weight');
      let sgender = await AsyncStorage.getItem('gender');
      let maritalstatus = await AsyncStorage.getItem('maritalstatus');
      let sblood = await AsyncStorage.getItem('bloodgroup');

      biostate(sbio);
      contactstate(scontact);
      agestate(sage);
      weightstate(sweight);
      genderstate(sgender);
      msstate(maritalstatus);
      bloodstate(sblood);
    }

    getuserdetail();
  });

  return (
    <View style={styles.sd}>
      <View style={styles.sdContainer}>
        <View style={styles.sdSubContainer}>
          <Icon size={18} name="user" color="#3743ab"></Icon>
        </View>
        <View style={styles.sdText}>
          <Text style={styles.sdText1}>{bio}</Text>
        </View>
      </View>
      <View style={styles.sdContainer}>
        <View style={styles.sdSubContainer}>
          <Icon size={18} name="phone-alt" color="#3743ab"></Icon>
        </View>
        <View style={styles.sdText}>
          <Text style={styles.sdText1}>{contact}</Text>
        </View>
      </View>
      <View style={styles.sdContainer}>
        <View style={styles.sdSubContainer}>
          <OctIcon size={18} name="number" color="#3743ab"></OctIcon>
        </View>
        <View style={styles.sdText}>
          <Text style={styles.sdText1}>{age + ' yrs'}</Text>
        </View>
      </View>
      <View style={styles.sdContainer}>
        <View style={styles.sdSubContainer}>
          <Icon size={18} name="weight" color="#3743ab"></Icon>
        </View>
        <View style={styles.sdText}>
          <Text style={styles.sdText1}>{weight + ' kg'}</Text>
        </View>
      </View>
      <View style={styles.sdContainer}>
        <View style={styles.sdSubContainer}>
          <IonIcon size={18} name="male-female" color="#3743ab"></IonIcon>
        </View>
        <View style={styles.sdText}>
          <Text style={styles.sdText1}>{gender}</Text>
        </View>
      </View>
      <View style={styles.sdContainer}>
        <View style={styles.sdSubContainer}>
          <Icon size={18} name="ring" color="#3743ab"></Icon>
        </View>
        <View style={styles.sdText}>
          <Text style={styles.sdText1}>{ms}</Text>
        </View>
      </View>
      <View style={styles.sdContainer}>
        <View style={styles.sdSubContainer}>
          <EntIcon size={18} name="drop" color="#3743ab"></EntIcon>
        </View>
        <View style={styles.sdText}>
          <Text style={styles.sdText1}>{blood}</Text>
        </View>
      </View>
    </View>
  );
};

export default SavedDetails;
