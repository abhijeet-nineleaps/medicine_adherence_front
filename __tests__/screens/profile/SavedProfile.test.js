import { render } from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SavedDetails from '../../../src/screens/profile/SavedDetails';
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useFocusEffect: jest.fn(),
}));
jest.mock("@react-native-async-storage/async-storage", () => ({
  getItem: jest.fn(),
}));
describe('Click send image', () => {
  it('renders correctly', () => {
    const tree = render(<SavedDetails/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
