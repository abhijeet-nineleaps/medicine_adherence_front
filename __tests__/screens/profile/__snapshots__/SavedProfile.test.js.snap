// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click send image renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "marginLeft": 8,
      "marginTop": 10,
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgrey",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 0,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3743ab",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 17,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgrey",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 0,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3743ab",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 17,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgrey",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 0,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3743ab",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 17,
          }
        }
      >
         yrs
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgrey",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 0,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3743ab",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 17,
          }
        }
      >
         kg
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgrey",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 0,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3743ab",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 17,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgrey",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 0,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3743ab",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 17,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgrey",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 0,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3743ab",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Entypo",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 17,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
